### PR TITLE
Changes to load nodes in local zones and remote nodes based on connection latencies

### PR DESCRIPTION
**Pull request by jshaofuturewei**
1. Added region and availability zones to data stores

2. Added latency threshold to check if the data store is a remote one
For example, if a datastore connection latency is more than or equals to 100ms, it will be a remote store. Otherwise it is a local store

3. Updated replica numbers to local and remote ones
Based on the 2, we will choose local replica number stores for synchronization writes. For remote stores, we will chose remote replica number stores for asynchronization writes.

4. Updated the consistent hashing to fetch availability zones first and then the nodes in the availability zones for local stores.

5. Added unit test step to workflow

### CODE REVIEW

File: cmd/http/config.json
Comment (pdgetrf) [inline cmd/http/config.json:11]: RemoteStoreLatencyThreshold -> RemoteStoreLatencyThresholdInMillSec

File: cmd/http/config.json
Comment (pdgetrf) [inline cmd/http/config.json:5]: do we still need local and remote config?

Comment (jshaofuturewei) [reply cmd/http/config.json:5]: The definition is at line 10  "//" : "The threshold is to check if a store is located in a remote datacenter. It is the latency in millis seconds to set up connection",

As we discussed, we need to config the number of LOCAL nodes for synchronized  write and the REMOTE nodes for asynchronized write.

File: pkg/constants/availability_zone.go
Comment (h-w-chen) [inline pkg/constants/availability_zone.go:9]: better NOT to use consts here; rkv targets to work in other cloud env.

File: cmd/http/main.go
Comment (h-w-chen) [inline cmd/http/main.go:218]: suggest to name as local (or localNodes)

File: pkg/constants/availability_zone.go
Comment (jshaofuturewei) [reply pkg/constants/availability_zone.go:9]: The reason to add constants here is to validate if the availability zone is valid since it will be used in hashing. If we use other cloud envs, we can add more constants.

File: cmd/http/config.json
Comment (jshaofuturewei) [reply cmd/http/config.json]: Updated.

Comment (jshaofuturewei) [reply cmd/http/config.json]: Updated

File: cmd/http/main.go
Comment (jshaofuturewei) [reply cmd/http/main.go]: Updated

File: cmd/http/config.json
Comment (pdgetrf) [inline cmd/http/config.json:6]: can these numbers (local being 2 and remote being 1) be calculated in the code based on the store configuration and the threshold?

Comment (jshaofuturewei) [reply cmd/http/config.json:6]: No. We discussed before. For low latencies, we try to write fewer local stores. However, it might cause uneven distribution. For even distribution, we might need to write many stores, which might cause high latency.  Something like CAP, we can't get both low latency and even distribution. The thing here is to find a balance point.  The first step is to manually configured. Next we can add policy or even learning model to get the best result.

File: cmd/http/config.json
Comment (pdgetrf) [reply cmd/http/config.json:6]: discussed offline. no issue here.

File: pkg/config/config.go
Comment (pdgetrf) [inline pkg/config/config.go:78]: this "else" not needed as there's a "return" on line 77

**Review by pdgetrf — APPROVED** [review]
LGTM

I don't have any further comments. Please check with @h-w-chen and see if he has any comments before merging.

**Review by h-w-chen — COMMENTED** [review]
comments only feedback on coding style. As long as it has approval from other peer, feel free to check in.

File: pkg/config/config.go
Comment (jshaofuturewei) [reply pkg/config/config.go:78]: Updated.

File: test/consistent/chain/chain_test.go
Comment (pdgetrf) [inline test/consistent/chain/chain_test.go:71]: what's the reason of this change?

Comment (jshaofuturewei) [reply test/consistent/chain/chain_test.go:71]: Ran several times locally but failed once since the write is async, it might take more time than expected, especially in git servers.

**Pull request merged**
